src/main/java/uk/gov/hmcts/reform/sscs/service/OnlineHearingService.java

```diff
@@ -83,12 +98,59 @@ private AppealDetails convertAppealDetails(SscsCaseDetails sscsCaseDetails) {
+        Address address = null;
+        Optional<Contact> contact = Optional.empty();
+        Name name = null;
+        Map<UserType,Subscription> subscriptionsMap = null;
+        UserType type = null;
+
+
+        Subscriptions subscriptions = sscsCaseDetails.getData().getSubscriptions();
+        Map<UserType,Subscription> appellantSubscriptions = of(Pair.of(UserType.APPELLANT, subscriptions.getAppellantSubscription()),
+                Pair.of(UserType.APPOINTEE, subscriptions.getAppointeeSubscription()),
+                Pair.of(UserType.REP, subscriptions.getRepresentativeSubscription()),
+                Pair.of(UserType.SUPPORTER, subscriptions.getSupporterSubscription()),
+                Pair.of(UserType.JOINT_PARTY, subscriptions.getJointPartySubscription()))
+                .filter(p -> p.getLeft() != null && p.getRight() != null)
+                .filter(p -> p.getRight().getEmail() != null || p.getRight().getMobile() != null)
+                .collect(Collectors.toMap(Pair::getLeft, Pair::getValue));
+
+
+        if (onlyForAppellant || isSignInSubscription(appellantSubscriptions.values(), tya, email)) {
+            address =  sscsCaseDetails.getData().getAppeal().getAppellant().getAddress();
+            contact = Optional.ofNullable(sscsCaseDetails.getData().getAppeal().getAppellant().getContact());
+            name = sscsCaseDetails.getData().getAppeal().getAppellant().getName();
+            subscriptionsMap = appellantSubscriptions;
+            type = UserType.APPELLANT;
+        } else {
+            for (CcdValue<OtherParty> op : emptyIfNull(sscsCaseDetails.getData().getOtherParties())) {
+                Map<UserType,Subscription> otherPartySubscriptions = of(Pair.of(UserType.OTHER_PARTY, op.getValue().getOtherPartySubscription()),
+                        Pair.of(UserType.OTHER_PARTY_APPOINTEE, op.getValue().getOtherPartyAppointeeSubscription()),
+                        Pair.of(UserType.OTHER_PARTY_REP, op.getValue().getOtherPartyRepresentativeSubscription()))
+                        .filter(p -> p.getLeft() != null && p.getRight() != null)
+                        .filter(p -> p.getRight().getEmail() != null || p.getRight().getMobile() != null)
+                        .collect(Collectors.toMap(Pair::getLeft, Pair::getValue));
+                if (isSignInSubscription(otherPartySubscriptions.values(), tya, email)) {
+                    address =  op.getValue().getAddress();
+                    contact = Optional.ofNullable(op.getValue().getContact());
+                    name = op.getValue().getName();
+                    subscriptionsMap = otherPartySubscriptions;
+                    type = UserType.OTHER_PARTY;
+                    break;
+                }
+            }
+        }
+
+        assert address != null;
+        return populateUserDetails(type, name, address, contact, subscriptionsMap);
```

Comment:
Can you break each bit down of the monster method into smaller separate methods please? Make it readable.

Reply:
Done